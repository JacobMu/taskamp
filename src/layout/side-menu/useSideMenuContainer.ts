import { useCallback, useState } from "react";
import { ITEM_TRANSLATION_KEY } from "./SideMenuService";

interface HookResults {
  isDialogOpen: boolean;
  handleOpen: (button: string) => void;
  handleClose: () => void;
}

export const useSideMenuContainer = (): HookResults => {
  const [isDialogOpen, setDialogOpen] = useState(false);

  const handleOpen = useCallback(
    (buttonType: string) => {
      if (buttonType === ITEM_TRANSLATION_KEY.CREATE_TASK) {
        setDialogOpen(true);
        return;
      }
      setDialogOpen(false);
    },
    [setDialogOpen]
  );

  const handleClose = useCallback(() => setDialogOpen(false), [setDialogOpen]);

  return {
    handleClose,
    handleOpen,
    isDialogOpen,
  };
};
