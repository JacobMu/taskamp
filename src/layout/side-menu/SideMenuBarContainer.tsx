import React from "react";
import { Button, IconName } from "@blueprintjs/core";

import "./SideMenuBarView.scss";
import { useSideMenuContainer } from "./useSideMenuContainer";
import type { ButtonProps } from "./SideMenuService";
import { CreateNewTaskDialogContainer } from "../dialog/CreateNewTaskDialogContainer";

interface Props {
  buttons: ButtonProps[];
  dispatchNewCard(cardProps: any): void;
}

export const SideMenuBarContainer: React.FC<Props> = ({
  buttons,
  dispatchNewCard,
}) => {
  const { isDialogOpen, handleClose, handleOpen } = useSideMenuContainer();

  return (
    <div className="sideMenuBarPosition">
      <CreateNewTaskDialogContainer
        isDialogOpen={isDialogOpen}
        handleClose={handleClose}
        dispatchNewCard={dispatchNewCard}
      />
      {buttons.map(({ i18n, icon, key }) => (
        <Button
          large
          icon={icon as IconName}
          key={key}
          onClick={() => handleOpen(key)}
        >
          {i18n}
        </Button>
      ))}
    </div>
  );
};
