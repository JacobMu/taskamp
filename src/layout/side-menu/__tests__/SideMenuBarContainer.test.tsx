import React from "react";
import { shallow } from "enzyme";
import { SideMenuBarContainer } from "../SideMenuBarContainer";
import { useSideMenuContainer } from "../useSideMenuContainer";
import {Button} from "@blueprintjs/core";

jest.mock("../useSideMenuContainer", () => ({
  useSideMenuContainer: jest.fn().mockReturnValue({
    isDialogOpen: false,
    handleClose: jest.fn().mockName("handleCloseMock"),
    handleOpen: jest.fn().mockName("handleOpenMock"),
  }),
}));

const DEFAULT_PROPS = {
  buttons: [{ i18n: "translation key", icon: "add", key: "key" }],
  dispatchNewCard: jest.fn().mockName("dispatchNewCardMock"),
};

describe("<SideMenuBarContainer />", () => {
  it("renders side menu bar", () => {
    const wrapper = shallow(<SideMenuBarContainer {...DEFAULT_PROPS} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("calls handleOpen with correct key as argument when clicked on button", () => {
    const handleOpenMock = jest.fn();
    (useSideMenuContainer as jest.Mock).mockReturnValue({
      handleOpen: handleOpenMock,
    });
    const wrapper = shallow(<SideMenuBarContainer {...DEFAULT_PROPS} />);

    wrapper.find(Button).simulate("click");

    expect(handleOpenMock).toHaveBeenCalledWith('key');
  });
});
