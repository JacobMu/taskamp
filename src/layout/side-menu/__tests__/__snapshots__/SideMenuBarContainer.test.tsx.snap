// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenuBarContainer /> renders side menu bar 1`] = `
<div
  className="sideMenuBarPosition"
>
  <CreateNewTaskDialogContainer
    dispatchNewCard={[MockFunction dispatchNewCardMock]}
    handleClose={[MockFunction handleCloseMock]}
    isDialogOpen={false}
  />
  <Blueprint3.Button
    icon="add"
    key="key"
    large={true}
    onClick={[Function]}
  >
    translation key
  </Blueprint3.Button>
</div>
`;
